test/ipv4or6/src/main.rs

```diff
@@ -0,0 +1,116 @@
+    for pipeline in pipelines {
+        sched.add_task(pipeline).unwrap();
+    }
+}
+
+fn main() {
+    let mut opts = basic_opts();
+    opts.optopt(
+        "",
+        "dur",
+        "Test duration",
+        "If this option is set to a nonzero value, then the \
+         test will exit after X seconds.",
+    );
+
+    let args: Vec<String> = env::args().collect();
+    let matches = match opts.parse(&args[1..]) {
+        Ok(m) => m,
+        Err(f) => panic!(f.to_string()),
+    };
+
+    let configuration = read_matches(&matches, &opts);
+
+    let test_duration: u64 = matches
+        .opt_str("dur")
+        .unwrap_or_else(|| String::from("0"))
+        .parse()
+        .expect("Could not parse test duration");
+
+    match initialize_system(&configuration) {
+        Ok(mut context) => {
+            context.start_schedulers();
+            context.add_pipeline_to_run(Arc::new(move |p, s: &mut StandaloneScheduler| test(p, s)));
+            context.execute();
+
+            if test_duration != 0 {
+                thread::sleep(Duration::from_secs(test_duration));
+            } else {
+                let mut pkts_so_far = (0, 0);
+                let mut last_printed = 0.;
+                const MAX_PRINT_INTERVAL: f64 = 30.;
+                const PRINT_DELAY: f64 = 15.;
+                let sleep_delay = (PRINT_DELAY / 2.) as u64;
+                let mut start = time::precise_time_ns() as f64 / CONVERSION_FACTOR;
+                let sleep_time = Duration::from_millis(sleep_delay);
+                println!("0 OVERALL RX 0.00 TX 0.00 CYCLE_PER_DELAY 0 0 0");
+                loop {
+                    thread::sleep(sleep_time); // Sleep for a bit
+                    let now = time::precise_time_ns() as f64 / CONVERSION_FACTOR;
+                    if now - start > PRINT_DELAY {
+                        let mut rx = 0;
+                        let mut tx = 0;
+                        for port in context.ports.values() {
+                            for q in 0..port.rxqs() {
+                                let (rp, tp) = port.stats(q);
+                                rx += rp;
+                                tx += tp;
+                            }
+                        }
+                        let pkts = (rx, tx);
+                        let rx_pkts = pkts.0 - pkts_so_far.0;
+                        if rx_pkts > 0 || now - last_printed > MAX_PRINT_INTERVAL {
+                            println!(
+                                "{:.2} OVERALL RX {:.2} TX {:.2}",
+                                now - start,
+                                rx_pkts as f64 / (now - start),
+                                (pkts.1 - pkts_so_far.1) as f64 / (now - start)
+                            );
+                            last_printed = now;
+                            start = now;
+                            pkts_so_far = pkts;
+                        }
+                    }
+                }
```

Comment:
nbd, but we can lose this whole big chunk if we just put a default dur on there. If we think it'd be worthwhile to set up a "listen forever and just handle/dump whatever comes through" though, I'm fine with leaving all this.

Reply:
sure.